cumulusci/tasks/datadictionary.py

```diff
@@ -0,0 +1,281 @@
+
+
+class GenerateDataDictionary(BaseGithubTask):
+    task_docs = """
+    Generate a data dictionary for the project by walking all GitHub releases.
+    The data dictionary is output as two CSV files.
+    One, in `object_path`, includes the Object Name, Object Label, and Version Introduced,
+    with one row per packaged object.
+    The other, in `field_path`, includes Object Name, Field Name, Field Label, Field Type,
+    Picklist Values (if any), Version Introduced.
+    """
+
+    task_options = {
+        "object_path": {
+            "description": "Path to a CSV file to contain an sObject-level data dictionary."
+        },
+        "field_path": {
+            "description": "Path to a CSV file to contain an field-level data dictionary."
+        },
+    }
+
+    def _init_options(self, kwargs):
+        super()._init_options(kwargs)
+
+        if self.options.get("object_path") is None:
+            self.options[
+                "object_path"
+            ] = f"{self.project_config.project__name} sObject Data Dictionary.csv"
+
+        if self.options.get("field_path") is None:
+            self.options[
+                "field_path"
+            ] = f"{self.project_config.project__name} Field Data Dictionary.csv"
+
+        if not self.project_config.project__git__prefix_release:
+            raise CumulusCIException(
+                "Set the `project__git__prefix_release` property in `cumulusci.yml` to use this task"
+            )
+
+    def _run_task(self):
+        self.logger.info("Starting data dictionary generation")
+
+        self._init_schema()
+        self._walk_releases()
+        self._write_results()
+
+    def _init_schema(self):
+        self.schema = defaultdict(
+            lambda: {"version": None, "fields": defaultdict(lambda: {"version": None})}
+        )
+
+    def _walk_releases(self):
+        repo = self.get_repo()
+
+        for release in repo.releases():
+            # Skip this release if any are true:
+            # It is a draft release
+            # It is prerelease (managed beta)
+            # This release's tag does not have the expected prefix,
+            # meaning we don't know its version number
+            if (
+                release.draft
+                or release.prerelease
+                or not release.tag_name.startswith(
+                    self.project_config.project__git__prefix_release
+                )
+            ):
+                continue
+
+            zip_file = download_extract_github_from_repo(repo, ref=release.tag_name)
+            version = self._version_from_tag_name(release.tag_name)
+            self.logger.info(f"Analyzing version {version}")
+
+            if "src/objects/" in zip_file.namelist():
+                # MDAPI format
+                self._process_mdapi_release(zip_file, version)
+
+            if "force-app/main/default/objects/" in zip_file.namelist():
+                # FIXME: check sfdx-project.json for directories to process.
```

Comment:
I've got some code in the retrieve-changes-better branch which does this too. We should probably turn it into a method of BaseProjectConfig.

Reply:
I'd like to reserve that improvement (along with handling dependencies) for the next version, by which time retrieve-changes-better I assume will be merged.